godot-core/src/builtin/vectors/vector_macros.rs

```diff
@@ -348,139 +363,300 @@ macro_rules! impl_float_vector_component_fns {
         ($($comp:ident),*)
     ) => {
         impl $Vector {
-            pub fn lerp(self, other: Self, weight: $Scalar) -> Self {
-                Self::new(
+            /// Returns a vector with the given components.
+            pub const fn new($($comp: $Scalar),*) -> Self {
+                Self {
                     $(
-                        self.$comp.lerp(other.$comp, weight)
+                        $comp
                     ),*
-                )
+                }
```

Comment:
```rs
$(
    $comp
),*
```
could now be written on a single line:
```rs
$( $comp ),*
```

Also in other similar places, unless they have super complex expressions within `$( ... )`

I think it helps readability, because the 4 levels of indentation are also quite distracting (`fn`, `Self {`, `$(`, expression).